bin/bookkeeper-daemon.sh

```diff
@@ -108,74 +108,101 @@ rotate_out_log ()
-    if [ -f $pid_file ]; then
-      TARGET_PID=$(cat $pid_file)
-      if kill -0 $TARGET_PID > /dev/null 2>&1; then
-        echo stopping $command
-        kill $TARGET_PID
-
-        count=0
-        location=$BOOKIE_LOG_DIR
-        while kill -0 $TARGET_PID > /dev/null 2>&1;
-        do
-          echo "Shutdown is in progress... Please wait..."
-          sleep 1
-          count=$(expr $count + 1)
-
-          if [ "$count" = "$BOOKIE_STOP_TIMEOUT" ]; then
-            break
+stop()
+{
+  if [ -f $pid_file ]; then
+        TARGET_PID=$(cat $pid_file)
+        if kill -0 $TARGET_PID > /dev/null 2>&1; then
+          echo stopping $command
+          kill $TARGET_PID
+
+          count=0
+          location=$BOOKIE_LOG_DIR
+          while kill -0 $TARGET_PID > /dev/null 2>&1;
+          do
+            echo "Shutdown is in progress... Please wait..."
+            sleep 1
+            count=$(expr $count + 1)
+
+            if [ "$count" = "$BOOKIE_STOP_TIMEOUT" ]; then
+              break
+            fi
+           done
+
+          if [ "$count" != "$BOOKIE_STOP_TIMEOUT" ]; then
+            echo "Shutdown completed."
           fi
-         done
-
-        if [ "$count" != "$BOOKIE_STOP_TIMEOUT" ]; then
-          echo "Shutdown completed."
-        fi
 
-        if kill -0 $TARGET_PID > /dev/null 2>&1; then
-          fileName=$location/$command.out
-          $JAVA_HOME/bin/jstack $TARGET_PID > $fileName
-          echo Thread dumps are taken for analysis at $fileName
-          if [ "$1" == "-force" ]
-          then
-            echo forcefully stopping $command
-            kill -9 $TARGET_PID >/dev/null 2>&1
-            echo Successfully stopped the process
-          else
-            echo "WARNNING :  Bookie Server is not stopped completely."
-            exit 1
+          if kill -0 $TARGET_PID > /dev/null 2>&1; then
+            fileName=$location/$command.out
+            $JAVA_HOME/bin/jstack $TARGET_PID > $fileName
+            echo Thread dumps are taken for analysis at $fileName
+            if [ "$1" == "-force" ]
+            then
+              echo forcefully stopping $command
+              kill -9 $TARGET_PID >/dev/null 2>&1
+              echo Successfully stopped the process
+            else
+              echo "WARNNING :  Bookie Server is not stopped completely."
+              exit 1
+            fi
           fi
+        else
+          echo no $command to stop
         fi
+        rm $pid_file
       else
         echo no $command to stop
       fi
-      rm $pid_file
-    else
-      echo no $command to stop
-    fi
+}
+case $startStop in
+  (start)
+    start "$*"
+    ;;
+
+  (stop)
+    stop $1
     ;;
+  (restart)
+      forceStopFlag=$(echo "$*"|grep "\-force")
+      if [[ "$forceStopFlag" != "" ]]
+      then
+        stop "-force"
+      else
+        stop
+      fi
+      if [ "$?" == 0 ]
+      then
+         sleep 3
+         paramaters="$*"
+         startParamaters=${paramaters//-force/}
```

Comment:
just a nit: if one of the command we have to start here accepts a flag like `-force-xxx` it will be blanked as well